Backend/src/controllers/auth.controller.ts

```diff
@@ -0,0 +1,102 @@
+import { Request, Response } from 'express';
+import bcrypt from 'bcryptjs';
+import jwt from 'jsonwebtoken';
+import { User, UserInput } from '../models/user.model';
+import { Token } from '../models/token.model';
+
+const generateToken = (userId: string, secret: string, expiresIn: string) => {
+  return jwt.sign({ id: userId }, secret, { expiresIn });
+};
+
+const register = async (req: Request, res: Response) => {
+  const { email, password, fullName, role } = req.body;
+  if (!email || !fullName || !password) {
+    return res.status(422).json({ message: 'The fields email, fullNamea and password are required' });
+  }
+
+  try {
+    const user = await User.findOne({ email });
+    if (user) {
+      return res.status(400).send('User already exist');
+    }
+
+    const hashedPassword = await bcrypt.hash(password, 10);
+    const newUser: UserInput = {
+      email,
+      password: hashedPassword,
+      enabled: true,
+      fullName,
+      role: role,
+    };
+    const userCreated = await User.create(newUser);
+    res.status(201).json({ name: userCreated.fullName, email: userCreated.email, _id: userCreated.id });
+  } catch (err) {
+    res.status(500).send('Server error');
+  }
+};
+
+const login = async (req: Request, res: Response) => {
+  const { email, password } = req.body;
+  try {
+    const user = await User.findOne({ email });
+    if (!user) {
+      return res.status(400).send('Invalid Email credentials');
+    }
+    const isMatch = await bcrypt.compare(password, user.password);
+    if (!isMatch) {
+      return res.status(400).send('Invalid Password credentials');
+    }
+
+    // delete other tokens that belogs to this user
+    await Token.deleteMany({ userId: user._id });
+
+    const accessToken = generateToken(user.id, 'access_token_secret', '1d');
+    const refreshToken = generateToken(user.id, 'refresh_token_secret', '7d');
+
+    const tokenDoc = new Token({
+      userId: user._id,
+      accessToken: accessToken,
+      refreshToken: refreshToken,
+      expiryDate: new Date(Date.now() + 7 * 24 * 60 * 60 * 1000),
+    });
+    await tokenDoc.save();
+
+    res.json({ accessToken, refreshToken });
+  } catch (err) {
+    res.status(500).send('Server error');
```

Comment:
Are we sending this error to client side?

Anyway we need to log this error details at the server side so that it will enable for identifying the issue.

Reply:
logged the error in server side

Reply:
Ok